Extensions/Physics3DBehavior/PhysicsCar3DRuntimeBehavior.ts

```diff
@@ -0,0 +1,1114 @@
+      implements gdjs.Physics3DRuntimeBehavior.BodyUpdater
+    {
+      carBehavior: gdjs.PhysicsCar3DRuntimeBehavior;
+      physicsBodyUpdater: gdjs.Physics3DRuntimeBehavior.BodyUpdater;
+
+      constructor(
+        carBehavior: gdjs.PhysicsCar3DRuntimeBehavior,
+        physicsBodyUpdater: gdjs.Physics3DRuntimeBehavior.BodyUpdater
+      ) {
+        this.carBehavior = carBehavior;
+        this.physicsBodyUpdater = physicsBodyUpdater;
+      }
+
+      private getVec3(x: float, y: float, z: float): Jolt.Vec3 {
+        const tempVec3 = this.carBehavior._sharedData._tempVec3;
+        tempVec3.Set(x, y, z);
+        return tempVec3;
+      }
+
+      createAndAddBody(): Jolt.Body | null {
+        const physics3D = this.carBehavior.getPhysics3D();
+        if (!physics3D) {
+          return null;
+        }
+        const { behavior } = physics3D;
+        const { _sharedData } = this.carBehavior;
+
+        const carShape = behavior.createShape();
+
+        // Create car body
+        const carBodySettings = new Jolt.BodyCreationSettings(
+          carShape,
+          this.carBehavior.getPhysicsPosition(_sharedData.getRVec3(0, 0, 0)),
+          behavior.getPhysicsRotation(_sharedData.getQuat(0, 0, 0, 1)),
+          Jolt.EMotionType_Dynamic,
+          behavior.getBodyLayer()
+        );
+        carBodySettings.mFriction = behavior.friction;
+        carBodySettings.mRestitution = behavior.restitution;
+        carBodySettings.mLinearDamping = behavior.linearDamping;
+        carBodySettings.mAngularDamping = behavior.angularDamping;
+        carBodySettings.mGravityFactor = behavior.gravityScale;
+        if (behavior.massOverride > 0) {
+          carBodySettings.mOverrideMassProperties =
+            Jolt.EOverrideMassProperties_CalculateInertia;
+          carBodySettings.mMassPropertiesOverride.mMass = behavior.massOverride;
+        }
+        const carBody = _sharedData.bodyInterface.CreateBody(carBodySettings);
+        Jolt.destroy(carBodySettings);
+        _sharedData.bodyInterface.AddBody(
+          carBody.GetID(),
+          Jolt.EActivation_Activate
+        );
+
+        // Create vehicle constraint
+        const vehicle = new Jolt.VehicleConstraintSettings();
+        vehicle.mUp = this.getVec3(0, 0, 1);
+        vehicle.mForward = this.getVec3(1, 0, 0);
+        vehicle.mMaxPitchRollAngle = gdjs.toRad(
+          this.carBehavior._pitchRollAngleMax
+        );
+
+        const FL_WHEEL = 0;
+        const FR_WHEEL = 1;
+        const BL_WHEEL = 2;
+        const BR_WHEEL = 3;
+        {
+          const setupWheel = (wheelS: Jolt.WheelSettingsWV) => {
+            wheelS.mWheelUp = this.getVec3(0, 0, 1);
+            wheelS.mWheelForward = this.getVec3(1, 0, 0);
+            wheelS.mSuspensionDirection = this.getVec3(0, 0, -1);
+            wheelS.mSteeringAxis = this.getVec3(0, 0, 1);
+          };
+
+          vehicle.mWheels.clear();
+
+          const fl = new Jolt.WheelSettingsWV();
```

Comment:
Can you confirm if this must be deleted or not? (same for fr/bl/br?)

Reply:
(i guess not as attached to mWheels, but please add a comment for this, like "ownership is handled by vehicule.mWheels") (or destroy if not)